### PR TITLE
676538 add edit setfield value

### PR DESCRIPTION
**Pull request by Manish-Josh-pepperi**

### CODE REVIEW

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 1]
```diff
@@ -1,15 +1,197 @@
...
+
+  async run() {
+    const filteredConfig = this._currentConfiguration.filter(config => config.Source !== "");
+    filteredConfig.forEach(config => this.resourceSet.add(config.UIListUUID));
+    this.listDataSource = this.getDataSource(filteredConfig); 
+    this.isLoaded = true;
+  }
+
+  addSourceCB = async (data: SetFieldValueConfiguration) => {
+    if (data && !this.resourceSet.has(data.UIListUUID)) {
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:56]: why do you need the resourceSet? why not add the "key" to the configuration object itself.. also, didn't we say you will generate a uuid?

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts [hunk 2]
```diff
@@ -1,15 +1,197 @@
...
+    this.hostObject.Configuration = this._currentConfiguration;
+  };
+
+  reload() {
+    const filteredConfig = this._currentConfiguration.filter(config => config.Source !== "");
+    this.listDataSource = this.getDataSource(filteredConfig);
+  }
+
+  add() {
+    this.hostObject.Configuration = this._currentConfiguration;
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.component.ts:81]: why do you need the current configuration on the "Add" dialog?

File: shared/set-field-value.model.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export interface SetFieldValueConfiguration {
+  Source: SourceType;
+  SourceValue: string;
+  FlowParam?: string;
+  PathParams?: IFieldConfig[];
+  DestinationObject: DestinationObjectType;
+  DestinationField: string;
+  DefaultValue?: string;
+  // this(UIListKey) is required for Ui to identified quniue item from pep-generic list edit/delete operation
+  UIListUUID?: string
```
Comment (roiaharon-pepperi) [inline shared/set-field-value.model.ts:19]: this is not a good property name.. this should be Key.

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.service.ts [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+
+  async getDestinationOptionsByType(
+    type: DestinationObjectType
+  ): Promise<IPepOption[]> {
+    // there is two approches
+    // 1. expose an EP to server side to getFieldsBytype() : as reference into story itself.
+    // 2. as shown below directly call the fields api from UI and pupolate the fields on UI.( check this is fine or not?)
+
+    const destinationFields: IPepOption[] = [];
+    this.getPapiCall(`/meta_data/${type}/fields`).subscribe((fields) => {
```
Comment (roiaharon-pepperi) [inline client-side/src/app/logic-blocks/set-field-value/set-field-value.service.ts:34]: this call should be on the server side, not the client side.. there shouldn't be any external call other that your server side

File: server-side/services/utilities-service.ts
Comment (roiaharon-pepperi) [inline server-side/services/utilities-service.ts:1]: what is this file? why do you need it?

File: server-side/services/transactions-service.ts
Comment (roiaharon-pepperi) [inline server-side/services/transactions-service.ts:1]: what is this file? why do you need it?

File: server-side/api.ts [hunk 5]
```diff
@@ -1,7 +1,11 @@
 import { Client, Request } from '@pepperi-addons/debug-server'
+import { TransactionsService } from './services/transactions-service'
+import { ApiFieldObject } from '@pepperi-addons/papi-sdk';
 
-export async function test(client: Client, request: Request): Promise<{Hello: string}> {
-    return {
-        Hello: 'World'
-    }
+export async function get_destination_fields(client: Client, request: Request): Promise<ApiFieldObject[]> {
+    const service = new TransactionsService(client, ''); // check if we can get the atdUUID from UI, for now we dont have one
```
Comment (roiaharon-pepperi) [inline server-side/api.ts:6]: you don't have an AtdUUID because you're not in the context of ATD..

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/logic-blocks/set-field-value/set-field-value.component.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+.configuration-content {
```
[the remark this answers is not in the document]
Comment (Manish-Josh-pepperi) [reply client-side/src/app/logic-blocks/set-field-value/set-field-value.component.scss:1]: No, we need this as the dialog size is not enough for the initial box.

**Comment by Manish-Josh-pepperi** [comment]
Closing this Pr as it contains unrequired commits.

**Pull request closed without merging**